optuna/visualization/matplotlib/_slice.py

```diff
@@ -1,11 +1,152 @@
+        sorted_params = sorted(list(set(params)))
+
+    n_params = len(sorted_params)
+
+    # Set up the graph style.
+    cmap = plt.get_cmap("Blues")
+    padding_ratio = 0.05
+    plt.style.use("ggplot")  # Use ggplot style sheet for similar outputs to plotly.
+
+    # Prepare data.
+    obj_values = [t.value for t in trials]
+
+    if n_params == 1:
+        # Set up the graph style.
+        fig, axs = plt.subplots()
+        axs.set_title("Slice Plot")
+
+        # Draw a scatter plot.
+        sc = _generate_slice_subplot(
+            trials, sorted_params[0], axs, cmap, padding_ratio, obj_values  # type: ignore
+        )
+    else:
+        # Set up the graph style.
+        min_figwidth = matplotlib.rcParams["figure.figsize"][0] / 2
+        fighight = matplotlib.rcParams["figure.figsize"][1]
+        # Ensure that each subplot has a minimum width without relying on auto-sizing.
+        fig, axs = plt.subplots(
+            1, n_params, sharey=True, figsize=(min_figwidth * n_params, fighight)
+        )
+        fig.suptitle("Slice Plot")
+
+        # Draw scatter plots.
+        for i, param in enumerate(sorted_params):
+            ax = axs[i]
+            sc = _generate_slice_subplot(
+                trials, param, ax, cmap, padding_ratio, obj_values  # type: ignore
+            )
+
+    axcb = fig.colorbar(sc, ax=axs)
+    axcb.set_label("#Trials")
+
+    return axs
+
+
+def _generate_slice_subplot(
+    trials: List[FrozenTrial],
+    param: str,
+    ax: "Axes",
+    cmap: "Colormap",
+    padding_ratio: float,
+    obj_values: List[Union[int, float]],
+) -> "PathCollection":
+    x_values = []
+    y_values = []
+    trial_numbers = []
+    for t, obj_v in zip(trials, obj_values):
+        if param in t.params:
+            x_values.append(t.params[param])
+            y_values.append(obj_v)
+            trial_numbers.append(t.number)
+    ax.set(xlabel=param, ylabel="Objective Value")
+    if _is_log_scale(trials, param):
+        ax.set_xscale("log")
+        xlim = _calc_lim_with_padding(x_values, padding_ratio, True)
+        ax.set_xlim(xlim[0], xlim[1])
+    else:
+        xlim = _calc_lim_with_padding(x_values, padding_ratio)
+        ax.set_xlim(xlim[0], xlim[1])
+    sc = ax.scatter(x_values, y_values, c=trial_numbers, cmap=cmap, edgecolors="grey")
+    ax.label_outer()
+
+    return sc
+
+
+def _calc_lim_with_padding(
+    values: List[Union[int, float]], padding_ratio: float, is_log_scale: bool = False
+) -> Tuple[Union[int, float], Union[int, float]]:
+    value_max = max(values)
+    value_min = min(values)
+    if is_log_scale:
+        padding = (math.log10(value_max) - math.log10(value_min)) * padding_ratio
+        return (
+            math.pow(10, math.log10(value_min) - padding),
+            math.pow(10, math.log10(value_max) + padding),
+        )
+    else:
+        padding = (value_max - value_min) * padding_ratio
```

Comment:
I think we have to consider categorical parameters here. ☝️ 
ref. https://colab.research.google.com/drive/1jcUmtMTuv9USSHoxCM6deUm3ojAlooCR?usp=sharing

Reply:
I agree with @himkt. Categorical parameters accept `None, bool, int, float, str`, so we need to take care of them in this function or in `_generate_slice_subplot` (I'm not sure if `ax.scatter` can handle `None` or `str`).

https://github.com/optuna/optuna/blob/965a3f31a7984d83ca1246a3554a77e9cf7370ed/optuna/distributions.py#L12

Reply:
Thank you for your suggestions, @himkt and @toshihikoyanase!
I'll work on that change this weekend.

Reply:
@himkt @toshihikoyanase 
Could you review the codes I added for handling categorical parameters?